source/hkdf.c

```diff
@@ -29,7 +29,18 @@ static const EVP_MD *aws_cryptosdk_get_evp_md(enum aws_cryptosdk_sha_version whi
     }
 }
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+// AWS-LC does not support HKDF construction with the |EVP_PKEY*| methods in OpenSSL 1.1.1.
+// We're backporting to ESDK's HKDF self implementation methods |aws_cryptosdk_hkdf_extract|
+// and |aws_cryptosdk_hkdf_expand| when building with AWS-LC for now.
+// The semantics of AWS-LC's HKDF API implementations (|HKDF_extract| & |HKDF_expand|)
+// are nearly identical to ESDK's. We can look to add support if there are customer
+// requirements to use AWS-LC's HKDF crypto implementation when building with ESDK.
```

Comment:
Can you clarify what you mean by the semantics of AWS-LC's HKDF API implementations being "nearly identical" to the ESDK's?